src/injectProps.js

```diff
@@ -1,8 +1,8 @@
 function injectProps(target, name, descriptor) {
   const oldFunction = descriptor.value;
 
-  descriptor.value = function propsInjectorFunction(...args) {
-    return oldFunction.bind(this)(...args, this.props);
+  descriptor.value = function propsInjectorFunction() {
```

Comment:
In that case I would suggest that the props are passed always as the first argument as otherwise the argument index would vary depending on the number of arguments which the function was invoked with.